### DISCUSSION TITLE
Background image with tailwind

### DISCUSSION THREAD

**Discussion by ShaungBhone** [tailwindlabs/tailwindcss#11515] [Help] [answered]
The tailwind background image is not shown on my web page. I'm using tailwind bg-[url] with laravel vite asset. The image is not working.

`<section class="bg-[url('{{ Vite::image('hero.jpg') }}')] bg-cover bg-no-repeat bg-blend-multiply">
`

So, I changed it like this. It's working.

`<section style="background-image: url('{{ Vite::image('hero.jpg') }}')"
`

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by ShaungBhone** [reply]
Thank you.

**Reply by abr4xas** [reply]
@wongjn ok, but what does your response have to do with the question? I'm having the same problem right now, and honestly, I can't find the similarity between what you're saying and how to solve it.

My issue:

I'm working with laravel and I have something like this:

```html
<div class="bg-[url('{{ $tour->getFirstMediaUrl('featured') }}')]"></div>
```

```html

<div class="bg-[url('https://cdn.somedomain.com/content/934/01J3JRM1RAKS41GZQK81WC682E.jpg')]"></div>
```

And even though I get the URL of my image, it doesn't display...

**Reply by wongjn** [reply]
My response quotes a guideline highlighted in the documentation that is violated in the OP. This violation of the guideline is what was causing the unexpected behavior.

This is the same as your given case. You are constructing a class name dynamically, and thus it does not work.

**Reply by abr4xas** [reply]
The documentation says that arbitrary values can be used (https://tailwindcss.com/docs/background-image#arbitrary-values). If I need an arbitrary image that can be uploaded by my user, does that mean I can't use it? :/

**Reply by wongjn** [reply]
You can:

```html
<div class="bg-[--image]" style="--image: url('{{ $tour->getFirstMediaUrl('featured') }}')"></div>
```

But you might as well then do:

```html
<div style="background-image: url('{{ $tour->getFirstMediaUrl('featured') }}')"></div>
```

**Reply by abr4xas** [reply]
This is how I solve my issue:

```html
<div
    class="bg-[image:var(--bg-image)]"
    style="
        --bg-image: url('{{ $tour->getFirstMediaUrl('featured') }}');
    "
>...</div>
```

Thanks @wongjn for your help 😃

**Comment by abdul-aziz-apu** [comment]
worked for me.

`<div class="bg-[url({{ asset('storage/images/banner_book_appoint.png') }})]"`

**Reply by RobinMalfait** [reply]
This can only work based on pure luck. 

Tailwind CSS scans classes as literal strings in your codebase. It doesn't have a runtime and it doesn't evaluate your code. This therefore can only work if the resolved class at runtime happens to exist in your codebase as-is or if you safelisted it manually.

More info: https://tailwindcss.com/docs/detecting-classes-in-source-files#dynamic-class-names

**Comment by abdul-aziz-apu** [comment]
I have added tailwindcss from the cdn on top of the html header section.
